python/semantic_kernel/core_plugins/crew_ai/crew_ai_enterprise.py

```diff
@@ -0,0 +1,267 @@
+        Returns:
+            str: The result of the task.
+
+        Raises:
+            FunctionExecutionException: If the task fails or an error occurs while waiting for completion.
+        """
+        try:
+            status_response: CrewAIStatusResponse | None = None
+            state: str = CrewAIEnterpriseKickoffState.Pending
+
+            async def poll_status():
+                nonlocal state, status_response
+                while state not in [
+                    CrewAIEnterpriseKickoffState.Failed,
+                    CrewAIEnterpriseKickoffState.Failure,
+                    CrewAIEnterpriseKickoffState.Success,
+                    CrewAIEnterpriseKickoffState.Not_Found,
+                ]:
+                    logger.debug(
+                        f"Waiting for CrewAI Crew with kickoff Id: {kickoff_id} to complete. Current state: {state}"
+                    )
+
+                    await asyncio.sleep(self.polling_interval)
+                    status_response = await self.client.get_status(kickoff_id)
+                    state = status_response.state
+
+            await asyncio.wait_for(poll_status(), timeout=self.polling_timeout)
+
+            logger.info(f"CrewAI Crew with kickoff Id: {kickoff_id} completed with status: {state}")
+            result = (
+                status_response.result if status_response is not None and status_response.result is not None else ""
+            )
+
+            if state in ["Failed", "Failure"]:
+                raise FunctionResultError(f"CrewAI Crew failed with error: {result}")
+            return result
+        except Exception as ex:
+            raise FunctionExecutionException(
+                f"Failed to wait for completion of CrewAI Crew with kickoff Id: {kickoff_id}."
+            ) from ex
+
+    def create_kernel_plugin(
+        self,
+        name: str,
+        description: str,
+        input_metadata: list[InputMetadata] | None = None,
+        task_webhook_url: str | None = None,
+        step_webhook_url: str | None = None,
+        crew_webhook_url: str | None = None,
+    ) -> KernelPlugin:
+        """Creates a kernel plugin that can be used to invoke the CrewAI Crew.
+
+        Args:
+            name (str): The name of the kernel plugin.
+            description (str): The description of the kernel plugin.
+            input_metadata (Optional[List[InputMetadata]], optional): The definitions of the Crew's
+            required inputs. Defaults to None.
+            task_webhook_url (Optional[str], optional): The task level webhook URL. Defaults to None.
+            step_webhook_url (Optional[str], optional): The step level webhook URL. Defaults to None.
+            crew_webhook_url (Optional[str], optional): The crew level webhook URL. Defaults to None.
+
+        Returns:
+            dict[str, Any]: A dictionary representing the kernel plugin.
+        """
+
+        def build_metadata(input_metadata: InputMetadata) -> KernelParameterMetadata:
```

Comment:
wait, we are creating KernelParameterMetadata here anyway, why not just use that instead of a new concept `InputMetadata`?